commit 88bf5a85fe9840c9b49c5f6c625cdccd11233943
Author: Sudeep Holla <sudeep.holla@arm.com>
Date:   Mon May 17 16:54:58 2021 +0100

    dt-bindings: dvfs: Add support for generic performance domains
    
    The CLKSCREW attack [0] exposed security vulnerabilities in energy management
    implementations where untrusted software had direct access to clock and
    voltage hardware controls. In this attack, the malicious software was able to
    place the platform into unsafe overclocked or undervolted configurations. Such
    configurations then enabled the injection of predictable faults to reveal
    secrets.
    
    Many Arm-based systems used to or still use voltage regulator and clock
    frameworks in the kernel. These frameworks allow callers to independently
    manipulate frequency and voltage settings. Such implementations can render
    systems susceptible to this form of attack.
    
    Attacks such as CLKSCREW are now being mitigated by not having direct and
    independent control of clock and voltage in the kernel and moving that
    control to a trusted entity, such as the SCP firmware or secure world
    firmware/software which are to perform sanity checking on the requested
    performance levels, thereby preventing any attempted malicious programming.
    
    With the advent of such an abstraction, there is a need to replace the
    generic clock and regulator bindings used by such devices with a generic
    performance domains bindings.
    
    [0] https://www.usenix.org/conference/usenixsecurity17/technical-sessions/presentation/tang
    
    Cc: Rob Herring <robh+dt@kernel.org>
    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Sudeep Holla <sudeep.holla@arm.com>
    Reviewed-by: Rob Herring <robh@kernel.org>
    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>

diff --git a/Documentation/devicetree/bindings/arm/cpus.yaml b/Documentation/devicetree/bindings/arm/cpus.yaml
index f3c7249c73d6..9a2432a88074 100644
--- a/Documentation/devicetree/bindings/arm/cpus.yaml
+++ b/Documentation/devicetree/bindings/arm/cpus.yaml
@@ -257,6 +257,13 @@ properties:
 
       where voltage is in V, frequency is in MHz.
 
+  performance-domains:
+    maxItems: 1
+    description:
+      List of phandles and performance domain specifiers, as defined by
+      bindings of the performance domain provider. See also
+      dvfs/performance-domain.yaml.
+
   power-domains:
     description:
       List of phandles and PM domain specifiers, as defined by bindings of the